File: 2.GammaCategorization/CreationalPatterns/5.Singleton/Aula07_SingletonExcercise/Aula_08_SinlgetonExercise2/C#/BOMBEIRO.cs
using System.Collections.Generic;

namespace CentralDeAlertasNamespace
{
    public class BOMBEIRO
    {
        public void EnviarAlerta(string mensagem)
        {
            CentralDeAlertas.Instancia.EnviarAlerta("BOMBEIRO", mensagem);
        }

        public IReadOnlyCollection<string> ListarAlertas()
        {
            return CentralDeAlertas.Instancia.GetAlertas();
        }
    }
}

File: 2.GammaCategorization/CreationalPatterns/5.Singleton/Aula07_SingletonExcercise/Aula_08_SinlgetonExercise2/BOMBEIRO.java
import java.util.List;

public class BOMBEIRO {
    public void enviarAlerta(String mensagem) {
        CentralDeAlertas.getInstancia().enviarAlerta("BOMBEIRO", mensagem);
    }

    public List<String> listarAlertas() {
        return CentralDeAlertas.getInstancia().getAlertas();
    }
}